ehrapy/anndata/_feature_specifications.py

```diff
@@ -0,0 +1,97 @@
+from typing import Literal
+
+import numpy as np
+import pandas as pd
+from rich import print
+from rich.tree import Tree
+
+from ehrapy import logging as logg
+from ehrapy.anndata._constants import CATEGORICAL_TAG, CONTINUOUS_TAG, DATE_TAG, FEATURE_TYPE_KEY
+from ehrapy.anndata.anndata_ext import anndata_to_df
+
+
+def infer_feature_types(adata, layer: str | None = None, output: Literal["print", "dataframe"] | None = "print"):
+    """
+    Infer feature types from AnnData object.
+
+    For each feature in adata.var_names, the method infers one of the following types: 'date', 'categorical', or 'continuous'.
+    The inferred types are stored in adata.var['feature_type']. Please check the inferred types and adjust if necessary using
+    adata.var['feature_type']['feature1']='corrected_type'.
+
+    Args:
+        adata: :class:`~anndata.AnnData` object storing the EHR data.
+        layer: The layer to use from the AnnData object. If None, the X layer is used.
+        output: The output format. Choose between 'print', 'dataframe', or None. If 'print', the feature types will be printed to the console.
+            If 'dataframe', a pandas DataFrame with the feature types will be returned. If None, nothing will be returned. Defaults to 'print'.
+    """
+    feature_types = {}
+
+    df = anndata_to_df(adata, layer=layer)
+    for feature in adata.var_names:
+        col = df[feature].dropna()
+        majority_type = col.apply(type).value_counts().idxmax()
+        if majority_type == pd.Timestamp:
```

Comment:
I sincerely hope that this catches all weird date formats that people can use, but it's probably Occam's razor

Reply:
It should work for all pandas dates. However, dates simply saved as a plain string are another thing... Probably the only way to check for those is to actually test for all possible date-string formats? #701 includes a ToDo to improve that. Or do you want this to be added in this PR?

Reply:
Couldn't stress test this yet against the approach TableOne has for feature type detection [here](https://github.com/tompollard/tableone/blob/bfd6fbaa4ed3e9f59e1a75191c6296a2a80ccc64/tableone/tableone.py#L555), but if it works for you fine so far we can go with that

Reply:
TableOne uses a completely different but also interesting approach by simply looking at the proportion of unique values for each feature. We could even think of combining the different approaches in the future -  I'll add a note to #701. I don't think that would solve the date issue though, e.g. because a birth date feature would usually have many different unique values.